app/models/song.rb

```diff
@@ -16,15 +16,35 @@ class Song < ActiveRecord::Base
   validates :chord_sheet, presence: true
   validates_inclusion_of :key, in: VALID_KEYS, allow_nil: true
   validates_inclusion_of :tempo, in: VALID_TEMPOS, allow_nil: true
-  before_save :normalize
+  before_save :normalize, :extract_lyrics
 
   def to_s
     name
   end
 
   private
+  # Titlize important fields and strip away unnecessary spaces
   def normalize
-    self.name = self.name.titleize
-    self.artist = self.artist.titleize if self.artist
+    self.name = self.name.titleize.strip
+    self.artist = self.artist.titleize.strip if self.artist
+    
+    normalized_lines = []
+    self.chord_sheet.split("\n").each { |line| normalized_lines << line.rstrip }
+    self.chord_sheet = normalized_lines.join("\n")
+  end
+
+  def extract_lyrics
+    lyrics = self.chord_sheet.split("\n").find_all { |line| is_line_of_lyrics(line) }
+    self.lyrics = lyrics.join("\n")
+  end
+
+  CHORD_GIVEAWAY_STRINGS = Set.new(%w(# / [ 2 3 4 7 9))
```

Comment:
If this was at work, I would say move the constant to the top of the private section then. But we're really obsessive about clean code and consistent formatting so up to you.